server/src/main/java/org/elasticsearch/action/support/tasks/TransportTasksAction.java

```diff
@@ -85,66 +81,164 @@ protected TransportTasksAction(
 
     @Override
     protected void doExecute(Task task, TasksRequest request, ActionListener<TasksResponse> listener) {
-        new AsyncAction(task, request, listener).start();
-    }
 
-    private void nodeOperation(CancellableTask task, NodeTaskRequest nodeTaskRequest, ActionListener<NodeTasksResponse> listener) {
-        TasksRequest request = nodeTaskRequest.tasksRequest;
-        processTasks(request, ActionListener.wrap(tasks -> nodeOperation(task, listener, request, tasks), listener::onFailure));
+        final var taskResponses = new ArrayList<TaskResponse>();
```

Comment:
Does it make sense to pre-size `taskResponses` based on the amount of the nodes from which we collect responses?

Reply:
I don't think the number of nodes would be a particularly useful estimate for this - as you say, many nodes return nothing, but sometimes they will return thousands of results. These are pretty low-throughput APIs so I think the default sizing is ok.